modules/virt-configuring-masquerade-mode-dual-stack.adoc

```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-configuring-masquerade-mode-dual-stack_{context}"]
+= Configuring masquerade mode with dual-stack (IPv4 and IPv6)
+
+You can configure a new virtual machine to use both IPv6 and IPv4 on the default pod network by using cloud-init.
+
+The IPv6 network address must be statically set to `fd10:0:2::2/120` with a default gateway of `fd10:0:2::1` in the virtual machine configuration. These are used by the virt-launcher pod to route IPv6 traffic to the virtual machine and are not used externally. 
+
+Once configured, incoming and outgoing traffic for the virtual machine is routed to both the IPv4 address and the unique IPv6 address of the virt-launcher pod.
+The virt-launcher pod then routes the IPv4 traffic to the DHCP address of the virtual machine, and the IPv6 traffic to the statically set IPv6 address of the virtual machine.
+
+.Prerequisites
+
+* The underlying network for the cluster must be connected with IPv6.
```

Comment:
One thing just occurred to me; do we want to explain how can this step be checked ?

Reply:
Yeah, we don't describe how to complete prerequisites. Sometimes they are assumed knowledge or previous procedures in an assembly. In certain cases we can xref to the related content, but not always (due to build limitations, or if the content does not exist in our openshift repo). 
How would a user verify this? Would it be on the openshift or the coreos level? I was working on the assumption that a network admin would know about the underlying network but maybe that's a bad assumption

Reply:
I'm not even sure if this is a prerequisite. VMs care only about IPv6 being available in OVN (inside the overlay). That can be done even if the underlying cluster has nodes connected over IPv4.

Our prerequisite is that OVN Kubernetes runs in dual-stack configuration